cmd/agent/serial.go

```diff
@@ -108,20 +108,44 @@ func RunSerialTerminalAgent(dest string, conf Config) error {
 	if err != nil {
 		return err
 	}
-	go func() {
+	go func(serial sr.Port) {
 		defer serial.Close()
 		if err := runSerialTerminalAgent(serial, conf); err != nil {
-			log.Fatalf("serial port error: %s", err)
+			log.Printf("serial port error: %s", err)
 		}
-	}()
+	}(serial)
 	return nil
 }
 
+// Read a line from the serial terminal.
+func readLine(reader io.Reader) (string, error) {
+	var buffer bytes.Buffer
+	buf := make([]byte, 1)
+	for {
+		n, err := reader.Read(buf)
+		if err != nil {
+			return buffer.String(), err
+		}
+		if n <= 0 {
+			continue
+		}
+		// Line breaks or termination symbol
+		if buf[0] == 13 || buf[0] == '\n' || buf[0] == '\000' {
```

Comment:
I think `'\r'` is more readable:

```suggestion
		if buf[0] == '\r' || buf[0] == '\n' || buf[0] == '\000' {
```

Reply:
Makes sense